agent/agentproc/proc.go

```diff
@@ -0,0 +1,127 @@
+package agentproc
+
+import (
+	"errors"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/spf13/afero"
+	"golang.org/x/xerrors"
+)
+
+const DefaultProcDir = "/proc"
+
+type Process struct {
+	Dir     string
+	CmdLine string
+	PID     int32
+	FS      afero.Fs
+}
+
+func (p *Process) SetOOMAdj(score int) error {
```

Comment:
This is all very unix and yet the file builds on all platforms. I don't think it's worth supporting priority management for the other (lesser) operating systems, but I think we should tag this unix for consistency with the wider repo.